src/services/ViewService.js

```diff
@@ -171,22 +170,29 @@ function ViewService($q, $http, $rootScope, URLService, ElementService, UtilsSer
                     }
                 }
             }
-            if (view.specification) {
+            if (view.specification && view.specification.operand) {
                 var specContents = view.specification.operand;
                 for (var j = 0; j < specContents.length; j++) {
                     if (specContents[j] && specContents[j].instanceId) {
                         toGet.push(specContents[j].instanceId);
                     }
                 }
             }
-            toGetSet = new Set(toGet);
+            if (isTable(view) && view.specification && view.specification.value) {
+                try {
+                    var tableJson = JSON.parse(view.specification.value);
+                    if (tableJson.body) {
+                        collectTableSources(toGet, tableJson.body);
+                    }
+                } catch (e) {
+                }
+            }
             $http.get(URLService.getViewElementIdsURL(reqOb))
             .then(function(response) {
                 var data = response.data.elementIds;
-                for (var i = 0; i < data.length; i++) {
-                    toGetSet.add(data[i]);
-                }
+                toGet = toGet.concat(data);
             }).finally(function() {
+                var toGetSet = new Set(toGet);
```

Comment:
I don't think you should reference toGet as you only assumed success in the 'then' clause.  This will mean that, regardless of success or failure of the promise to return, your finally is NOW dependent on success.  Based on the content of this finally clause, the code within it should probably be moved up, and the original $http.get call should be chained with the 'getElements' promise being made in the finally clause.  Of course, if the code below doesn't change, referencing a variable that may not exist on failure of the original promise is a moot point.  Or, I could be totally misunderstanding: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally

Reply:
toGet is declared in the higher context, if the get succeeds it just adds to it, otherwise we proceed with what we already had

Reply:
ok